arith/polynomials/src/univariate.rs

```diff
@@ -0,0 +1,222 @@
+use std::ops::{AddAssign, Index, IndexMut, MulAssign};
+
+use arith::{FFTField, Field};
+use itertools::izip;
+
+#[derive(Debug, Clone, Default)]
+pub struct UnivariatePoly<F: Field> {
+    pub coeffs: Vec<F>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct UnivariateLagrangePolynomial<F: FFTField> {
+    pub evals: Vec<F>,
+}
+
+impl<F: Field> Index<usize> for UnivariatePoly<F> {
+    type Output = F;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.coeffs[index]
+    }
+}
+
+impl<F: Field> IndexMut<usize> for UnivariatePoly<F> {
+    fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+        &mut self.coeffs[index]
+    }
+}
+
+impl<F: FFTField> Index<usize> for UnivariateLagrangePolynomial<F> {
+    type Output = F;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        &self.evals[index]
+    }
+}
+
+impl<F: FFTField> IndexMut<usize> for UnivariateLagrangePolynomial<F> {
+    fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+        &mut self.evals[index]
+    }
+}
+
+impl<F: Field> UnivariatePoly<F> {
+    #[inline]
+    pub fn new(coeffs: Vec<F>) -> Self {
+        Self { coeffs }
+    }
+
+    /// Sample a random polynomials.
+    #[inline]
+    pub fn random(degree: usize, mut rng: impl rand::RngCore) -> Self {
+        let coeffs = (0..=degree).map(|_| F::random_unsafe(&mut rng)).collect();
+        Self { coeffs }
+    }
+
+    /// Evaluation by Horner's rule
+    #[inline]
+    pub fn evaluate(&self, point: F) -> F {
+        let mut eval = F::ZERO;
+        self.coeffs.iter().rev().for_each(|t| {
+            eval *= point;
+            eval += *t
+        });
+        eval
+    }
+
+    #[inline(always)]
+    pub fn degree_one_quotient(&self, alpha: F) -> (Self, F) {
```

Comment:
nits: a short description of the function

Reply:
see https://github.com/PolyhedraZK/Expander/pull/225/commits/f0d9a0d77de7b2c4bd70e6c4dd7c835d115c7c6e